packages/wdio-reporter/src/index.ts

```diff
@@ -23,22 +45,8 @@ export default class WDIOReporter extends EventEmitter {
         this.outputStream = this.options.stdout || !this.options.logFile
             ? options.writeStream
             : fs.createWriteStream(this.options.logFile)
-        this.failures = []
-        this.suites = {}
-        this.hooks = {}
-        this.tests = {}
-        this.currentSuites = []
-        this.counts = {
-            suites: 0,
-            tests: 0,
-            hooks: 0,
-            passes: 0,
-            skipping: 0,
-            failures: 0
-        }
-        this.retries = 0
 
-        let currentTest
+        let currentTest: TestStats
 
         const rootSuite = new SuiteStats({
```

Comment:
This still produces an error, because there are quite a few non-optional properties defined in suite.ts for a Suite object that should be passed to the constructor of SuiteStats which are missing here.
Making all these missing properties optional is also not possible, because `uid` is used as index at line 71 & 162.
Please suggest how to solve this.